migrations/versions/0410_enums_for_everything.py

```diff
@@ -0,0 +1,1151 @@
+    BrandType: {
+        "values": ["govuk", "org", "both", "org_banner"],
+        "name": "brand_types",
+    },
+    InvitedUserStatus: {
+        "values": ["pending", "accepted", "cancelled", "expired"],
+        "name": "invited_user_statuses",
+    },
+    AuthType: {
+        "values": ["sms_auth", "email_auth", "webauthn_auth"],
+        "name": "auth_types",
+    },
+    JobStatus: {
+        "values": [
+            "pending",
+            "in progress",
+            "finished",
+            "sending limits exceeded",
+            "scheduled",
+            "cancelled",
+            "ready to send",
+            "sent to dvla",
+            "error",
+        ],
+        "name": "job_statuses",
+    },
+    NotificationStatus: {
+        "values": [
+            "cancelled",
+            "created",
+            "sending",
+            "sent",
+            "delivered",
+            "pending",
+            "failed",
+            "technical-failure",
+            "temporary-failure",
+            "permanent-failure",
+            "pending-virus-check",
+            "validation-failed",
+            "virus-scan-failed",
+        ],
+        "name": "notify_statuses",
+    },
+    NotificationType: {
+        "values": ["sms", "email", "letter"],
+        "name": "notification_types",
+    },
+    OrganizationType: {
+        "values": ["federal", "state", "other"],
+        "name": "organization_types",
+    },
+    CallbackType: {
+        "values": ["delivery_status", "complaint"],
+        "name": "callback_types",
+    },
+    ServicePermissionType: {
+        "values": [
+            "email",
+            "sms",
+            "international_sms",
+            "inbound_sms",
+            "schedule_notifications",
+            "email_auth",
+            "upload_document",
+            "edit_folder_permissions",
+        ],
+        "name": "service_permission_types",
+    },
+    RecipientType: {"values": ["mobile", "email"], "name": "recipient_types"},
+    TemplateType: {"values": ["sms", "email", "letter"], "name": "template_types"},
+    TemplateProcessType: {
+        "values": ["normal", "priority"],
+        "name": "template_process_types",
+    },
+    CodeType: {"values": ["email", "sms"], "name": "code_types"},
+}
+
+
+def enum_create(values: list[str], name: str) -> None:
+    enum_db_type = postgresql.ENUM(*values, name=name)
+    enum_db_type.create(op.get_bind())
+
+
+def enum_drop(values: list[str], name: str) -> None:
+    enum_db_type = postgresql.ENUM(*values, name=name)
+    enum_db_type.drop(op.get_bind())
+
+
+def enum_using(column_name: str, enum: Enum) -> str:
+    return f"{column_name}::text::{_enum_params[enum]['name']}"
+
+
+def enum_type(enum: Enum) -> sa.Enum:
+    return sa.Enum(
+        *_enum_params[enum]["values"],
+        name=_enum_params[enum]["name"],
+        values_callable=(lambda x: [e.value for e in x]),
+    )
+
+
+@contextmanager
+def view_handler() -> Iterator[None]:
```

Comment:
Oh nicely done for handling cascading changes and making sure everything gets sorted out before recreating the view again!